### DISCUSSION TITLE
rfd 15

### DISCUSSION THREAD

**Discussion by dudeisbrendan03** [gravitational/teleport#7765] [General]
https://github.com/gravitational/teleport/blob/master/rfd/0015-2fa-management.md

Any updates on multiple 2-factor auth methods, we're ordering in batches of security keys and moving people over from otp to u2f- but Teleport doesn't allow us to mix-n-match 2fa methods

**Comment by webvictim** [comment]
I believe it's currently possible for an individual user to add multiple second factors. For example, if the user has a TOTP second factor already configured, they can run `tsh mfa add --type=u2f` and add their U2F key - they'll be required to type a TOTP code to confirm it.

The other way round also works - run `tsh mfa add --type=totp` and authenticate this with an existing U2F key.

Once multiple devices are added, they should be able to use any of them to provide a second factor to Teleport.
